db.go

```diff
@@ -431,7 +431,11 @@ func QueryDeadlineMySQL57(db *DB) error {
 func ContextInfoRewriter(infoFromContext func(ctx context.Context) string) func(*DB) error {
 	return func(db *DB) error {
 		db.contextInfoRewriter = func(ctx context.Context, db *DB, query string) (queryWithInfo string, err error) {
-			query = fmt.Sprintf("/* %s */ %s", infoFromContext(ctx), query)
+			info := infoFromContext(ctx)
+			// Do not allow info if it would close the comment. This could be a SQL injection.
+			if !strings.Contains(info, "*/") {
```

Comment:
We should also disallow strings that begin with `!` (see https://dev.mysql.com/doc/refman/5.7/en/comments.html)

Reply:
I think it would be ok since there is a space after the start of the comment and before the info, but adding anyway in case some one changes that.